presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/QueryCompletedEvent.java

```diff
@@ -45,6 +48,7 @@ public QueryCompletedEvent(
         this.context = requireNonNull(context, "context is null");
         this.ioMetadata = requireNonNull(ioMetadata, "ioMetadata is null");
         this.failureInfo = requireNonNull(failureInfo, "failureInfo is null");
+        this.warnings = requireNonNull(warnings, "warnings is null");
```

Comment:
Uh, that's a pity. Can't we use guava _internally_ in SPI and shade it?

If we don't `copyOf(..)` then maybe `new ArrayList(..)`? Otherwise this is weak.